app/messages/isolation_segment_assign_org_message.rb

```diff
@@ -0,0 +1,34 @@
+require 'messages/base_message'
+
+module VCAP::CloudController
+  class IsolationSegmentAssignOrgMessage < BaseMessage
+    ALLOWED_KEYS = [:data].freeze
+
+    attr_accessor(*ALLOWED_KEYS)
+
+    def self.create_from_http_request(body)
+      IsolationSegmentAssignOrgMessage.new(body.symbolize_keys)
+    end
+
+    validates_with NoAdditionalKeysValidator
+    validates :data, presence: true, array: true, allow_nil: false, allow_blank: false
+    validates_each :data do |record, attr, values|
+      if values.is_a? Array
```

Comment:
This doesn't work as well, because it's a validates_each, not a simple validates - if we move this totally to a custom validator, that validator seems to get invoked before the `validates :data` one, which means we have to reimplement the type checking (for arrays), etc., so it's not really an improvement.  The only alternative I can suggest, if you're simply after simplification, is to move the body to a private method (e.g. `validates_each ... do |...| private_method(args) end`

>